frontend/src/app/shared/components/storages/storage/storage.component.ts

```diff
@@ -237,21 +237,27 @@ export class StorageComponent extends UntilDestroyedMixin implements OnInit, OnD
   ngOnInit():void {
     this.storage = this.storagesResourceService.requireEntity(this.projectStorage._links.storage.href);
 
-    this.fileLinks = this.collectionKey()
-      .pipe(
-        switchMap((key) => {
-          if (isNewResource(this.resource)) {
-            return this.fileLinkResourceService.collection(key);
-          }
+    this.fileLinks = this.storage.pipe(
```

Comment:
❓ What is the reasoning to put the whole collection load behind the storage observable?

Reply:
Note: storage observable loads auth state, then the race condition between those two requests is eliminated.